src/ModbusSlave.cpp

```diff
@@ -147,9 +147,17 @@ int Modbus::poll() {
     // check minimum length.
     if (lengthIn < 8) return 0;
 
-    // this removes the null character that sometimes
-    // is get from the SoftwareSerial on the "nano"
-    if(lengthIn > 8 && bufIn[lengthIn-1] == 0) lengthIn--;
+    /**
+     * this removes the null character that sometimes
+     * is get from the SoftwareSerial on the "nano"
+     * posible message length are:
+     *   FC1..6 : length = 8  (with extra null 9)
+     *   FC15   : length = 11 (with extra null 12)
```

Comment:
Ooo, this is very wrong ... will fix :-(

Reply:
will need to do crc after checking command and length ?

Reply:
I don't like hardcoding the length (as I also did in the first pull request).
If anybody will ever implement new messages that are part of the modbus standard, it will be a mess again.
I would prefer to isolate a function calculating the message length and trimming the nulls at the end.
Do you agree?

Reply:
👍 yes, fixed that ...